recipes/newrelic/infrastructure/ohi/memcached/debian.yml

```diff
@@ -81,49 +66,117 @@ install:
+
+          # Set default values for variables
+          NR_CLI_HOST=${NR_CLI_HOST:-localhost}
+          NR_CLI_PORT=${NR_CLI_PORT:-11211}
+          MAX_RETRIES=3
+          TRIES=0 
+          CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+          while [[ $TRIES -lt $MAX_RETRIES  && ( $CAN_CONNECT == "6" || $CAN_CONNECT == "7" ) ]]; do
+            if [ $CAN_CONNECT == "6" ]; then 
+              printf "\nMemcached Hostname (default: localhost): "
+              read -r NR_CLI_HOST
+              NR_CLI_HOST=${NR_CLI_HOST:-localhost}
+            fi
+
+            printf "\nMemcached Port (default: 11211): "
+            read -r NR_CLI_DB_PORT
+            NR_CLI_DB_PORT=${NR_CLI_DB_PORT:-11211}
+            ((TRIES++))
+
+            CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+            if [ $CAN_CONNECT == "6" ]; then
+              printf "\n[Error]: Provided host ($NR_CLI_DB_HOSTNAME) cannot be reached.\nSee https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/postgresql-monitoring-integration#comp-req for more info.\n" >&2
+              if [ ! $TRIES -lt $MAX_RETRIES ]; then exit 6; fi
+              echo "Please try again"
+            elif [ $CAN_CONNECT == "7" ]; then
+              printf "\n[Error]: Connections to port ($NR_CLI_DB_PORT) are being refused.\nSee https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/postgresql-monitoring-integration#comp-req for more info.\n" >&2
+              if [ ! $TRIES -lt $MAX_RETRIES ]; then exit 7; fi
+              echo "Please try again"
+            else
+              break
+            fi
+          done
+          
+          CAN_CONNECT=""
+          if [ -z "$NR_CLI_DB_USERNAME" ] && [ -z "$NR_CLI_DB_PASSWORD" ]; then
+            CAN_CONNECT=$(memcstat --servers="$NR_CLI_HOST" | grep uptime | wc -l)
+          else
+            CAN_CONNECT=$(memcstat --servers="$NR_CLI_HOST" --username="$NR_CLI_DB_USERNAME" --password="$NR_CLI_DB_PASSWORD" | grep uptime | wc -l)
+          fi
+
+          TRIES=0 
+          while [[ $TRIES -lt $MAX_RETRIES && $CAN_CONNECT -eq 0 ]]; do
+            printf "\nMemcached Username: "
+            read -r NR_CLI_DB_USERNAME
+            printf "Memcached Password: "
+            stty -echo
+            read -r NR_CLI_DB_PASSWORD
+            stty echo
+            ((TRIES++))
+
+            CAN_CONNECT=$(memcstat --servers="$NR_CLI_HOST" --username="$NR_CLI_DB_USERNAME" --password="$NR_CLI_DB_PASSWORD" | grep uptime | wc -l)
+            
+            if [ $CAN_CONNECT -eq 0 ] ; then
+              printf "\n\n[Error]: Memcached server is not accessible with the provided username, password.\nSee https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/postgresql-monitoring-integration#comp-req for more info.\n" >&3
```

Comment:
@NRhzhao similarly here, the [Error]: is a bit scary, Maybe we could rephrase to not be as alarming.

Reply:
Updated